### DISCUSSION TITLE
如何生成 0.5.4对应的 odrive_endpoints.hpp

### DISCUSSION THREAD

**Discussion by Singerwall** [Factor-Robotics/odrive_ros2_control#26] [Q&A]
你好  Yuan 

 该文件如何生成? 我使用的是 0.5.4的固件 , 无法通信. 因此我想 生成 0.5.4对应的 odrive_endpoints.hpp
https://github.com/Factor-Robotics/odrive_ros2_control/blob/foxy-fw-v0.5.3/odrive_hardware_interface/include/odrive_hardware_interface/odrive_endpoints.hpp

**Comment by borongyuan** [comment]
您好，在0.5.1之后，odrivetool generate-code就不再被支持。
https://github.com/odriverobotics/ODrive/issues/593
因此当我们适配0.5.3时用了一些繁琐的步骤。因为我们可用于维护该项目的时间不多，所以我们跳过了0.5.2（反正这版固件也有点毛病），也会跳过0.5.4（反正这版改动不大），之后会支持0.5.5。我们也在研究根据固件版本自动确定协议的实现方案，以免之后要维护更多分支。对于新固件，目前还是建议重刷到0.5.3或0.5.1使用。

**Comment by Singerwall** [comment]
好的,谢谢!
我在odriver 官网看到了不支持,但是给了 一些繁琐的步奏, 我整不会.&nbsp;&nbsp;
''在研究根据固件版本自动确定协议的实现方案''
&nbsp; &nbsp;ack: 在学习源码时 发现可以使用xxx.j2文件自动生成odrive_endpoints.hpp, 然后去找如何写xxx.j2 文件 ,又发现 xxx.j2文件也是自动生成的..... 就卡主了





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Factor-Robotics/odrive_ros2_control"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年10月18日(星期二) 中午11:02
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [Factor-Robotics/odrive_ros2_control] 如何生成 0.5.4对应的 odrive_endpoints.hpp (Discussion #26)





 
您好，在0.5.1之后，odrivetool generate-code就不再被支持。
 odriverobotics/ODrive#593
 因此当我们适配0.5.3时用了一些繁琐的步骤。因为我们可用于维护该项目的时间不多，所以我们跳过了0.5.2（反正这版固件也有点毛病），也会跳过0.5.4（反正这版改动不大），之后会支持0.5.5。我们也在研究根据固件版本自动确定协议的实现方案，以免之后要维护更多分支。对于新固件，目前还是建议重刷到0.5.3或0.5.1使用。
